src/main/java/com/biggestnerd/namecolors/NameColors.java

```diff
@@ -55,7 +55,9 @@ public void updatePlayerName(Player player, ChatColor color) {
 			CivChat2.getInstance().getCivChat2Manager().removeCustomName(player.getUniqueId());
 			if(getServer().getPluginManager().isPluginEnabled("TAB")) {
 				//TAB is enabled, so lets reset the player name.
-				getPlayer(player.getUniqueId()).removeTemporaryValue(EnumProperty.CUSTOMTABNAME);
+				if(getPlayer(player.getUniqueId()).hasTemporaryValue(EnumProperty.CUSTOMTABNAME)) {
+					getPlayer(player.getUniqueId()).removeTemporaryValue(EnumProperty.CUSTOMTABNAME);
```

Comment:
Could you please save those `getPlayer(player.getUniqueId())` to a variable so that it's not run twice?